pkg/cmd/cli/cmd/cancelbuild.go

```diff
@@ -105,7 +105,7 @@ func RunCancelBuild(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, arg
 	// Create a new build with the same configuration.
 	if cmdutil.GetFlagBool(cmd, "restart") {
 		request := &buildapi.BuildRequest{
-			ObjectMeta: kapi.ObjectMeta{Name: build.Name},
+			ObjectMeta: kapi.ObjectMeta{Name: build.Name, Namespace: namespace},
 		}
 		newBuild, err := client.Builds(namespace).Clone(request)
```

Comment:
> The namespace should be filled - this change shouldn't be necessary.

Filled where?  It really looked like the resource builder was doing it to me and if I don't set it, then `oc start-build` fails on: 
```Error from server: buildRequest "ruby-sample-build-invalidtag" is invalid: metadata.namespace: required value```